### PR TITLE
[W5.4][F11-A2]AN JIASHENG

### PR DESCRIPTION
**Pull request by AngularJiaSheng**

### CODE REVIEW

File: src/seedu/addressbook/data/Password/PasswordStorage.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public class PasswordStorage {
+    private static String actualPassword = "ABC123";
+    private static final String EXAMPLE_PASSWORD = "example_password";
+
+    public PasswordStorage(){
+
+    }
+
+    public static String getActualPassword(){
+        return actualPassword;
```
Comment (Choony93) [inline src/seedu/addressbook/data/Password/PasswordStorage.java]: Password storage are usually not stored in plaintext as this makes it exposed and anyone can see it. Consider reading up on password hashing to understand proper techniques in storing passwords without compromising it's confidentiality.

http://security.blogoverflow.com/2013/09/about-secure-password-hashing/

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -15,6 +11,10 @@
 import seedu.addressbook.storage.StorageFile.StorageOperationException;
 import seedu.addressbook.ui.TextUi;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
```
Comment (PierceAndy) [inline src/seedu/addressbook/Main.java]: Avoid making unrelated changes into your PR.

File: src/seedu/addressbook/commands/ViewAllCommand.java [hunk 3]
```diff
@@ -32,6 +36,9 @@ public CommandResult execute() {
             if (!addressBook.containsPerson(target)) {
                 return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
             }
+            if(!userInputPassword.equals(PasswordStorage.getActualPassword())){
```
Comment (PierceAndy) [inline src/seedu/addressbook/commands/ViewAllCommand.java]: Missing space before curly brace.

File: src/seedu/addressbook/commands/ViewAllCommand.java [hunk 4]
```diff
@@ -11,17 +12,20 @@
 public class ViewAllCommand extends Command {
 
     public static final String COMMAND_WORD = "viewall";
+    public String userInputPassword;
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Views the non-private details of the person "
```
Comment (PierceAndy) [inline src/seedu/addressbook/commands/ViewAllCommand.java]: The help message's description of this command does not match it's effect.

File: src/seedu/addressbook/data/Password/PasswordStorage.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    }
+    public static String getExamplePassword(){
```
Comment (PierceAndy) [inline src/seedu/addressbook/data/Password/PasswordStorage.java]: Missing new line between methods + missing space before opening curly brace.

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -1,35 +1,23 @@
...
-import static seedu.addressbook.common.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+import seedu.addressbook.commands.*;
+import seedu.addressbook.data.exception.IllegalValueException;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.*;
```
Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java]: Import statements should be listed explicitly.
See point 3 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#statements

File: src/seedu/addressbook/parser/Parser.java [hunk 7]
```diff
@@ -1,35 +1,23 @@
 package seedu.addressbook.parser;
 
-import static seedu.addressbook.common.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.addressbook.common.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+import seedu.addressbook.commands.*;
```
Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java]: Import statements should be listed explicitly.
See point 3 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#statements

File: src/seedu/addressbook/parser/Parser.java [hunk 8]
```diff
@@ -229,6 +220,38 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
...
+    /**
+     * Parses the given arguments string as a string of Password.
+     *
+     * @param args arguments string to parse as index number
+     * @return the parsed index number
+     * @throws ParseException if no region of the args string could be found for the index
+     * @throws NumberFormatException the args string region is not a valid number
+     * @throws IndexOutOfBoundsException if args.split()[] method cannot be carried out.
+     */
+    private String parseArgsAsPassword(String args) throws IndexOutOfBoundsException, ParseException, NumberFormatException{
```
Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java]: Keep lines no longer than 120 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.
See points 2 & 3 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#layout

Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java]: Keep lines no longer than 120 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.
See points 2 & 3 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#layout

File: src/seedu/addressbook/parser/Parser.java [hunk 9]
```diff
@@ -229,6 +220,38 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
...
+    /**
+     * Parses the given arguments string as a single index number.
+     *
+     * @param args arguments string to parse as index number for Viewall command specifcally
+     * @return the parsed index number
+     * @throws ParseException if no region of the args string could be found for the index
+     * @throws NumberFormatException the args string region is not a valid number
+     * @throws IndexOutOfBoundsException if " " input is passed in.
+     */
+    private int parseArgsAsDisplayedIndexForViewAll(String args) throws ParseException, NumberFormatException, IndexOutOfBoundsException {
```
Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java]: Keep lines no longer than 120 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.
See points 2 & 3 here: https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-Java.adoc#layout

File: src/seedu/addressbook/parser/Parser.java [hunk 10]
```diff
@@ -229,6 +220,38 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
         }
         return Integer.parseInt(matcher.group("targetIndex"));
     }
+    /**
```
Comment (PierceAndy) [inline src/seedu/addressbook/parser/Parser.java]: Missing blank line between methods.

**Comment by PierceAndy** [comment]
@AngularJiaSheng Some comments added. Please close the PR after reading comments.

File: src/seedu/addressbook/data/Password/PasswordStorage.java (hunk 1 above)
Comment (AngularJiaSheng) [reply src/seedu/addressbook/data/Password/PasswordStorage.java]: hey thank you so much! I will definitely check out that link:):)

File: src/seedu/addressbook/parser/Parser.java (hunk 8 above)
Comment (AngularJiaSheng) [reply src/seedu/addressbook/parser/Parser.java]: Thank you!
